SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -41,13 +41,27 @@ Interest InterestFactory::CreateInterest()
 		return Interest{};
 	}
 
-	if (Actor->GetNetConnection() != nullptr)
+	if (GetDefault<USpatialGDKSettings>()->bEnableServerQBI)
 	{
-		return CreatePlayerOwnedActorInterest();
+		if (Actor->GetNetConnection() != nullptr)
+		{
+			return CreatePlayerOwnedActorInterest();
+		}
+		else
+		{
+			return CreateActorInterest();
+		}
 	}
 	else
 	{
-		return CreateActorInterest();
+		if (Actor->IsA(APlayerController::StaticClass()))
+		{
+			return CreatePlayerOwnedActorInterest();
+		}
+		else
+		{
+			return Interest{};
```

Comment:
This will still attach an (empty) Interest component to all entities. Does that mean QBI is still used for those entities?

Reply:
More to the point -- will a component with no queries satisfy the performance concerns?

Reply:
I believe it doesn't but we can double check this.